src/objective.jl

```diff
@@ -160,3 +160,36 @@ function objective_function(model::Model,
                    MOI.ObjectiveFunction{MOIFunType}())::MOIFunType
     return jump_function(model, func)
 end
+
+"""
+    set_objective_coefficient(model::Model, variable::VariableRef, coefficient::Real)
+
+Set the linear objective coefficient associated with `Variable` to `coefficient`.
+
+Note: this function will throw an error if a nonlinear objective is set.
+"""
+function set_objective_coefficient(model::Model, variable::VariableRef, coeff::Real)
+    if model.nlp_data !== nothing && _nlp_objective_function(model) !== nothing
+        error("A nonlinear objective is already set in the model")
+    end
+
+    obj_fct_type = moi_function_type(objective_function_type(model))
+    if obj_fct_type == MOI.SingleVariable
+        # Promote the objective function to be an affine expression.
+        current_obj = objective_function(model)
+        if index(current_obj) == index(variable)
+            set_objective_function(model, coeff * variable)
+        else
+            set_objective_function(model, current_obj + coeff * variable)
```

Comment:
You will allocate two AffExpr here. Do `add_to_expression!(coeff * variable, current_obj)` to allocate only one